File: sdks/sdkwork-craw-chat-sdk-admin/sdkwork-craw-chat-sdk-admin-flutter/composed/lib/src/protocol_module.dart
import 'package:craw_chat_admin_backend_sdk/craw_chat_admin_backend_sdk.dart';

import 'context.dart';

class CrawChatAdminProtocolModule {
  final CrawChatAdminSdkContext context;

  CrawChatAdminProtocolModule(this.context);

  Future<JsonObject> getGovernance() {
    return context.backendClient.protocol.getProtocolGovernance();
  }

  Future<JsonObject> getRegistry() {
    return context.backendClient.protocol.getProtocolRegistry();
  }
}

File: sdks/sdkwork-craw-chat-sdk-admin/sdkwork-craw-chat-sdk-admin-typescript/composed/src/protocol-module.ts
import type { JsonObject } from './types.js';
import type { CrawChatAdminSdkContext } from './sdk-context.js';

export class CrawChatAdminProtocolModule {
  constructor(private readonly context: CrawChatAdminSdkContext) {}

  getGovernance(): Promise<JsonObject> {
    return this.context.backendClient.protocol.getProtocolGovernance();
  }

  getRegistry(): Promise<JsonObject> {
    return this.context.backendClient.protocol.getProtocolRegistry();
  }
}
